BestPractices.md

```diff
@@ -32,6 +32,55 @@ if item == nil {
 }
 ```
 
+### Assertion management ###
+
+When working with optional values you have to make sure that the variable you work with doesn’t have a `nil` value. To do so you can use the different unwrapping techniques provided by the standard library, but in certain scenarios it can make sense to force unwrap your value and terminate your program (for example a wrong view controller type after instantiating from a Storyboard). In this case your app better terminate in order to avoid unexpected behaviors.
+
+The standard Swift library provides you different assertion functions that affect your code differently:
+
+* assert
+* assertionFailure
+* precondition
+* preconditionFailure
+* fataError
+
+#### assert ####
+`assert` is only evaluated in debug mode, it means that the line will be removed in release and will not be executed.
+
+#### assertionFailure ####
+`assertionFailure` acts like `assert` but provides some context to the compiler.
+
+#### precondition ####
+`precondition` ensures that the given condition is meet. If not the app will terminate. `precondition` works for both debug and release.
+
+#### preconditionFailure ####
+`preconditionFailure` means a fatal error and will terminate in both debug and release mode, except for unchecked builds (`-Ounchecked`), then it will never be executed.
+
+#### fatalError ####
+`fatalError` acts like `preconditionFailure` but is not affected by the unchecked build flag. It will always terminate your app in both debug and release mode.
+
+#### Production ####
+
+Although crashing on production is not always the ideal. Often you prefer crashing on debug but in production handle the error with a default value like 0 or empty array. To do so, we recommand implementing a custom operator that takes an optional value as well as a tuple containing the default value of the same type and a error message for the context.
+
+```swift
+infix operator ?! {}
```

Comment:
well you know my thoughts on the custom operator. 

If we _have_ to have this, can we have the custom operator be a shortcut for a function call so people who don't want to use / do this (hello, myself) can instead use the function?